crypto/threads_win.c

```diff
@@ -27,6 +27,88 @@
 # define NO_INTERLOCKEDOR64
 #endif
 
+/* Check for older versions of MSVC compiler before version 1929 on x86 architecture. */
+#if defined(_MSC_VER) && defined(_M_IX86) && _MSC_VER < 1929
+
+/* Inline assembly for InterlockedAdd64 to add 64-bit integers atomically */
+static __declspec(naked) __int64 __fastcall InterlockedAdd64(__int64* volatile _Ptr, __int64 _Val)
+{
+    __asm {
+        push    ebx
+        push    esi
+        mov     esi, ecx          ; Pointer to the value
+        mov     eax, [esi]        ; Load low part of the value
+        mov     edx, [esi+4]      ; Load high part of the value
+    lbl: mov     ebx, eax
+        mov     ecx, edx
+        add     ebx, [esp+8+4]    ; Add low part of _Val
+        adc     ecx, [esp+8+8]    ; Add high part of _Val with carry
+        lock cmpxchg8b qword ptr [esi] ; Compare and exchange if unchanged
```

Comment:
Is this going to work for 32 bit windows platforms?  I feel like it might be safer for older platforms to use the fallback approach that pthreads uses (i.e. using an internal mutex)